src/main/java/com/google/gwt/dom/client/InputElement.java

```diff
@@ -381,6 +384,19 @@ public final void setValue(String value) {
   }-*/;
 
   /**
+   * @Mock
+   * 
+   * Set value in native element (like Selenium 'sendKeys').
+   * The new value will be available in calls to getValue().
+   * N.B. Will not trigger ValueChange events until focus lost.
+   */
+  public final void setNewValue(String value) {
+      focus();
+      setAttribute("mock.Value.new", value);
```

Comment:
I indeed to use this library for UI components testing.  eventually  I will move my components for JS implementations and will just test controller using this library. Sorry but I don't see the complete path yet, I just hope that I will be able to so.

So I think the problem I see is maybe common pattern of order of events handling. So other people will reuse the same approach or will make it better.
For now I'm not suggesting any implementation to this mock library for this, But I intend to do so soon once I do some research and find how to other people are testing the same.